Find a functionally equivalent code:com.android.internal.os.ZygoteConnection.handleChildProc:COMMENT
Method Modifier: private     internal    
Comment:/**
 * Handles post-fork setup of child proc, closing sockets as appropriate,
 * reopen stdio as appropriate, and ultimately throwing MethodAndArgsCaller
 * if successful or returning if failed.
 *
 * @param parsedArgs non-null; zygote args
 * @param descriptors null-ok; new file descriptors for stdio if available.
 * @param pipeFd null-ok; pipe for communication back to Zygote.
 * @param newStderr null-ok; stream to use for stderr until stdio
 * is reopened.
 *
 * @throws ZygoteInit.MethodAndArgsCaller on success to
 * trampoline to code that invokes static main.
 */

@@ -1,4 +1,9 @@
 {
+    /**
+     * By the time we get here, the native code has closed the two actual Zygote
+     * socket connections, and substituted /dev/null in their place.  The LocalSocket
+     * objects still need to be closed properly.
+     */
     closeSocket();
     ZygoteInit.closeServerSocket();
     if (descriptors != null) {
@@ -19,25 +24,30 @@
         if (parsedArgs.invokeWith != null) {
             WrapperInit.execApplication(parsedArgs.invokeWith, parsedArgs.niceName, parsedArgs.targetSdkVersion, pipeFd, parsedArgs.remainingArgs);
         } else {
-            RuntimeInit.zygoteInit(parsedArgs.targetSdkVersion, parsedArgs.remainingArgs);
+            RuntimeInit.zygoteInit(parsedArgs.targetSdkVersion, parsedArgs.remainingArgs, null);
         }
     } else {
         String className;

